README.md

```diff
@@ -119,3 +125,177 @@ yarn build-all
 ```
 
 Optimised production bundles for both `ui` and `legacy` will be built, and output to `./build`.
+
+## Testing with Cypress
+
+[Cypress](https://www.cypress.io/) is an end-to-end Javascript testing framework that executes in the browser, and therefore in the same run loop as the device under test. It includes features such as time travel (through the use of UI snapshots), real-time reloads and automatic/intuitive waiting.
+
+### Running headless tests
+
+To run headless Cypress tests, enter the following command from the root of the project:
+
+```
+yarn test-cypress
+```
+
+This will automatically start legacy, ui and proxy servers and run the Cypress tests, in which results are logged to the console. After running the tests, the servers and process will close.
+
+### Interactive testing
+
+By launching the Cypress Test Runner, you will be able to to see commands as they execute while also viewing the UI while it's being tested. Note that because the Cypress Test Runner is a graphical application, launching it in a container or VM will require some extra steps because you will need to forward the XVFB messages from Cypress out of the container into an X11 server running on the host machine.
+
+#### Interactive testing on host machine
+
+If developing directly on your host machine, simply run maas-ui in development as normal:
+
+```
+yarn serve
+```
+
+Then open the Cypress Test Runner by running:
+
+```
+yarn cypress-open
+```
+
+You should then see a list of test specs in maas-ui. You can run all interactive tests by clicking "Run all specs" in the top-right of the window.
+
+#### Interactive testing in LXD
```

Comment:
Maybe add something in this section about running into `max_user_watches` issues and linking to the solution: https://medium.com/ppl-c6-big-data/whats-wrong-with-my-simple-react-docker-image-4e81584dac1b